be/src/util/threadpool.cpp

```diff
@@ -444,6 +451,15 @@ bool ThreadPool::wait_for(const MonoDelta& delta) {
                                [&]() { return _total_queued_tasks <= 0 && _active_threads <= 0; });
 }
 
+void ThreadPool::update_max_threads(int max_threads) {
```

Comment:
No need to add lock?

Reply:
Not need that. Because we don't need this to be see immediately, and update int32 is nature atomic.

Reply:
In fact, it should be atomic to write an `int` in x86, because it will be align in cache line. But for more generic, I add std::atomic to protect it. Done.